extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -174,11 +176,13 @@ private void writeTaskPayload(Task task) throws IOException
    * @return
    * @throws IllegalStateException
    */
-  protected synchronized TaskStatus join(long timeout) throws IllegalStateException
+  protected synchronized TaskStatus join(long timeout, SettableFuture<Boolean> taskActiveStatusFuture) throws IllegalStateException
```

Comment:
This is an odd pattern; it's more typical for a method to return `ListenableFuture<T>`. If the caller needs to chain that onto a `SettableFuture<T>`, the caller should attach a callback that sets the settable future.

Please also add javadoc explaining what the returned future means.

Reply:
i guess it would make more sense to maybe split this up into a joinUntilStart method that calls back into the wait method, i can rewrite it a bit

Reply:
hmm doing it this way kinda requires the entire lifecycle logic to be rewritten, i think i have a simpler semi-hacky way of accomplishing the same thing.